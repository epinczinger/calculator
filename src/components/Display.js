import React from 'react';
import PropTypes from 'prop-types';

const Display = props => {
  const { next, total } = props;

  return (
    <div className="flex justify-end bg-gray-400 border-4 border-gray-800 rounded-md text-2xl lg-text-3xl px-2 py-3 ">
      <h2 className="text-gray-700">
        {next || total || 0}
      </h2>
    </div>
  );
};

Display.defaultProps = {
  next: '0',
  total: '',
};

Display.propTypes = {
  next: PropTypes.number,
  total: PropTypes.string,
};

export default Display;
