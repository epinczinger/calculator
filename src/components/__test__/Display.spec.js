import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../Display';

describe('The Display', () => {
  it('renders as expected matching snapshot', () => {
    const tree = renderer
      .create(<Display next={1} total="3" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
