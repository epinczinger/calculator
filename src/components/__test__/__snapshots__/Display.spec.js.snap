// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Display renders as expected matching snapshot 1`] = `
<div
  className="flex justify-end tracking-tighter bg-gray-400 border-4 w-64 border-gray-800 rounded-md text-lg lg:w-auto lg:text-2xl px-2 py-3 "
>
  <h2
    className="text-gray-700"
  >
    1
  </h2>
</div>
`;
